phalcon/di/service.zep

```diff
@@ -174,7 +174,13 @@ class Service implements ServiceInterface
 					 * Bounds the closure to the current DI
 					 */
 					if typeof dependencyInjector == "object" {
-						let definition = \Closure::bind(definition, dependencyInjector);
+						let reflectionFunction = new \ReflectionFunction(definition);
+    						if (
+							reflectionFunction->getClosureScopeClass() === null ||
+							reflectionFunction->getClosureThis() !== null
+    						) {
+							let definition = \Closure::bind(definition, dependencyInjector);
+						}
```

Comment:
in the case we've given a static closure (which can't be bound), the use of $this in the closure won't work (not really a pb as this shouldn't be used inside a closure...), but the system will continue fo function as it was in Phalcon 2.0.